pytorch_lightning/loggers/neptune.py

```diff
@@ -15,25 +15,72 @@
 Neptune Logger
 --------------
 """
+__all__ = [
+    "NeptuneLogger",
+]
+
 import logging
+import operator
+import os
 from argparse import Namespace
-from typing import Any, Dict, Iterable, Optional, Union
+from functools import reduce
+from typing import Any, Dict, Generator, Optional, Set, Union
+from weakref import ReferenceType
 
 import torch
-from torch import is_tensor
 
+from pytorch_lightning import __version__
 from pytorch_lightning.loggers.base import LightningLoggerBase, rank_zero_experiment
-from pytorch_lightning.utilities import _module_available, rank_zero_only
+from pytorch_lightning.utilities import rank_zero_only
+from pytorch_lightning.utilities.imports import _NEPTUNE_AVAILABLE, _NEPTUNE_GREATER_EQUAL_0_9
+from pytorch_lightning.utilities.model_summary import ModelSummary
+from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
+
+if _NEPTUNE_AVAILABLE and _NEPTUNE_GREATER_EQUAL_0_9:
+    try:
+        from neptune import new as neptune
+        from neptune.new.run import Run
+        from neptune.new.exceptions import NeptuneLegacyProjectException
+    except ImportError:
+        import neptune
+        from neptune.run import Run
+        from neptune.exceptions import NeptuneLegacyProjectException
+else:
+    # needed for test mocks, and function signatures
+    neptune, Run = None, None
+
 
 log = logging.getLogger(__name__)
-_NEPTUNE_AVAILABLE = _module_available("neptune")
 
-if _NEPTUNE_AVAILABLE:
-    import neptune
-    from neptune.experiments import Experiment
-else:
-    # needed for test mocks, these tests shall be updated
-    neptune, Experiment = None, None
+INTEGRATION_VERSION_KEY = "source_code/integrations/pytorch-lightning"
+
+# kwargs used in NeptuneLogger for legacy client (current NeptuneLegacyLogger)
+LEGACY_NEPTUNE_INIT_KWARGS = [
+    "project_name",
+    "offline_mode",
+    "experiment_name",
+    "experiment_id",
+    "params",
+    "properties",
+    "upload_source_files",
+    "abort_callback",
+    "logger",
+    "upload_stdout",
+    "upload_stderr",
+    "send_hardware_metrics",
+    "run_monitoring_thread",
+    "handle_uncaught_exceptions",
+    "git_info",
+    "hostname",
+    "notebook_id",
+    "notebook_path",
+]
+
+# kwargs used in legacy NeptuneLogger from neptune-pytorch-lightning package
+LEGACY_NEPTUNE_LOGGER_KWARGS = [
```

Comment:
if what @tchaton suggests isn't possible then at least we should make it protected :) 
```suggestion
_LEGACY_NEPTUNE_LOGGER_KWARGS = [
```